pkg/promtail/client/client.go

```diff
@@ -349,6 +356,9 @@ func (c *client) getTenantID(labels model.LabelSet) string {
 
 // Stop the client.
 func (c *client) Stop() {
+	// cancel any upstream calls made using client's `ctx`.
+	c.cancel()
+
 	c.once.Do(func() { close(c.quit) })
```

Comment:
Now you can get rid of c.quit by replacing the listener from 

```
	for {
		select {
		case <-c.quit:
			return
```

to 

```
	for {
		select {
		case <-c.ctx.Done():
			return
```

Reply:
@cyriltovena true!, but should we do it in this PR? I actually thought of just deprecating `quit` and remove it later completely. 
anything works for me!

Reply:
yep works for me too.